### ISSUE TITLE
Fuel vanishing in reference to issue #413

### ISSUE THREAD

**Issue by Gunner76th** [erogenousbeef/BigReactors#492]
This is a re-opening of issue #413 as linked here https://github.com/erogenousbeef/BigReactors/issues/413

tl;dr: Issue confirmed to be a result of having a reactor (or turbine) built over a smeltry at any level. Due to smeltry mechanics, at some point during chunk load, the smeltry is clearing the fluid inventory of the reactor/turbine. SlimeKnights has been linked into this bug report to help figure out a resolution, short of "Don't put anything with a fluid inventory above a smeltry".

So I have begun testing with one of the reactors on my server that a player is having issues with. At first I thought the issue was because 1/2 of the reactor was being chunk loaded while the other half was not. As a counter to this, I tried 2 different chunk loading methods, first I placed boilers from Flaxbeards Steam Power mod which does natural chunk loading (as in does not report as being force loaded, shows up green on opis) and did a log cycle. When I logged out the reactor was full, with 40 ingots in the port, after being logged out fro about 15 secs, I logged back in, and the port was down to 16 ingots. Thinking perhaps it was an issue with natural chunk loading when the player was not online, I placed chicken chunk loaders, 1 at each corner of the reactor (overkill I know, but I wanted to be damn sure it was force loaded) to keep the chunks force loaded (loaded chunks that show up as force loaded and blue in opis). Again, I completely filled up the reactor, and set the port to have 40 ingots in it. Again, I log out, wait about 15 seconds and log back in, and again, the port is down to 16 ingots in just 15-30 seconds of me being offline.  As a final test, I removed all chunk loading from the area, refilled the reactor, set the ingots to 40 in the port, and proceeded with my log cycle. This time the ingots depleted slower, with 32 ingots remaining instead of the previous 16. 

Included are photos of the reactors design, and as you can see, the only blocks touching the reactor itself is cobblestone and 2 chests, neither of which should be causing this issue. The chest above the fuel port is "attached" to a second fuel port set to exhaust mode, while the bottom port is set to input mode. The power is being removed via EnderIO Energy conduit. The reactor itself sits directly above a Tinkers Construct Smeltry, however it is 1 block space above the smeltry, so it is not physically touching it. As you can see, the inside of the reactor has no blocks or fluids at all. 

Playing on a hunch, which seems to have proven true, the cause of at least my personal issue, is the Smeltry itself. I know from personal experience, that once a smeltry is placed down, if you try to place a second smeltry above it, regardless of how high up you go, it will fail to form. This is because once it is formed, the smeltry reserves all block spaces above it in the event you decide to expand the smeltry upwards to increase your inventory capacity within the controller. Based on my understanding of the mechanics for the smeltry, I hypothisize that it is in some way causing the "fluid fuel" within the reactor to be cleared out to match the state of the smeltry, in essence it is treating the fluid storage portion of the reactor as if it was part of the smeltry. Because of this personal confirmation, I am going to be linking Boni into this bug report as well, since this is now an intermod incompatibility issue, and might possibly be something she is able to resolve on her end.

**Comment by Gunner76th** [comment]
Here are all of the screen shots I took during the testing. https://www.dropbox.com/sh/cogbs3ld90wm26d/AAD2UNaJjGtTH4b7zA1BS7wka?dl=0

**Comment by erogenousbeef-zz** [comment]
Hmm. Haven't had a chance to look at the screenshots yet, but this sounds promising.

However, _passively-cooled_ reactors do not use the fluid system for storage in 0.4; they use a custom "reactant" system which isn't exposed via any public interface - I actually marshal back and forth between solids and reactants, or fluids and reactants.

Turbines, however, do use the fluid tanks internally.

If the smeltery were doing something funky on chunk-load, that could certainly disrupt things, but it's still difficult to conceptualize. However, it's a worthy avenue of investigation. Can this be reproduced with the latest BR/latest TC in SSP?

**Comment by Gunner76th** [comment]
It will take me a while to setup and test with the latest BR/TCon, but I could certainly do it. I will need to setup a forge 1387 server, and client pairing. I could perhaps have results by Sunday at the latest.

**Comment by erogenousbeef-zz** [comment]
This issue has been open for so long, as few more days won't kill anyone; I appreciate the enthusiasm and willingness to help. Would you mind testing SSP before trying SMP, though? It's generally easier to iterate on fixes in SSP, to singleplayer reproductions are far better for me than SMP.

**Comment by Gunner76th** [comment]
Yes I can do that fairly easy and that would prolly make it easier to setup on my end as well. I will try my best to enable all debuging for both mods as well, incase that reveals anything that you might be able to understand.

**Comment by Gunner76th** [comment]
Ok, I have an SSP world setup, Super flat, with NEI, COFH Core, Big Reactors, TCon, and Mantle, most recent versions of all mods, as well as Forge 1387 using the Mojang launcher. I will begin testing tomorrow, and should have some results by this weekend. If you by chance have dropbox, I can create a share collaboration folder that I can give you access to, and then I can just drop all screen shots and log files into that and it will sync directly to your computer in real time also. Otherwise, I can just drop share links on here to the files.

**Comment by Gunner76th** [comment]
Ok, so far in testing in SSP, I have 3 identical reactors setup, 1 as a control with no smeltry, 1 with a smeltry above it, and 1 with a smeltry below it. So far, when I save and quit the session, then I start the session back up, there is no loss of fuel. I also went about 400 blocks away and waited about 10-15 secs to allow the area to unload then returned back. When I returned back, again, there was no loss of fuel. 

In SSP mode, with the most recent versions of Big Reactors, Mantle, TCon, and  Forge 1387, the issue of vanishing fuel seems to not be present.

I need to push this world over to my server, and setup an SMP session to test that as well, to make sure it isnt an issue that is isolated to SMP worlds. 

I have a feeling that with the changes to the fluid registry in forge 1384+ as well as whatever changes TCon made between 1.8.3b and 1.8.4a seems to have resolved this issue, in regards to SSP worlds. SMP still requires further testing.

**Comment by Gunner76th** [comment]
Also, as an extra test of things, and because of this being how it was setup on my live server, all 3 reactors, as well as the 2 smeltries sat on the corner where 4 chunks met.

**Comment by andykorth** [comment]
I've also had an issue with vanishing fuel in my big reactor. We're running FTB Infinity 1.3.0 in SMP which includes Big Reactors 0.4.2A2.

Whenever I change dimensions or log out when my big reactor is running, when I return to it, the fuel is empty, and no cyanite was generated. My smeltery is several chunks away, but I do have a multiblock railcraft iron tank above the reactor that feeds water into the reactor. Occasionally, when I return the multiblock tank is not a proper miultiblock structure, and I need to break and replace a block on it for it to return. Even if that doesn't occur, the big reactor is empty of fuel. I wonder if the fluid storage of the iron tank causes a similar problem to the smeltery. 

I have no smeltery in the immediate area and the iron tanks aren't touching the reactor.

I have a second iron tank above my BR turbine that holds steam. The turbine seems to suffer no ill effects.

Hope that helps. Thanks!

**Comment by Aiur850** [comment]
I'm also having the same issue on a server I play on.

If my reactor is chunk loaded and I sign off, it's fine. But once I remove the chunk loader and let the chunks unload, either by signing off or moving a few hundred blocks away and return, I find my fuel has vanished, including any excess ingots sitting in the access port.

Seems to me the issue is mainly happening with chunks being loaded/unloaded. I can't provide any more information though as I'm not admin on the server and don't have access to the configs of the server for debugging.

**Comment by andykorth** [comment]
I recreated my setup in the single player version and wasn't able to replicate the issue. _shrugs_

**Comment by ghost** [comment]
I've been able to replicate the issue. It has something to do with the loading and unloading of chunks.

**Comment by Armigus** [comment]
Hard confirmation that this issue is persistent and in SSP.  I have a massive reactor that has been losing fuel repeatedly and unexpectedly.  Capacity is 5,424 ingots, then depletion reduces it to 1,104.  Finding this issue led me to test with chicken chunks chunk loader.  That prevented the depletion.

I tested by visiting another dimension and then returning.  Evidently the block data is not being stored properly to prevent loss upon chunk unloading.  Fix this now... I have likely lost almost a million yellorium to this glitch and want no repeat losses.

**Comment by XFactHD** [comment]
First of all, don't demand fixes, the authors are doing this for fun and they also have a RL.
Secondly, Mr E. Beef has dropped this project because of missing time. The 1.7.10 version was continued by CyberdyneCC. You might want to download the latest version provided there and test again.
I have already looked into this and have not yet found any apparent reason for this issue.

**Comment by Gunner76th** [comment]
@XFactHD do you have the link for the new authors page for Big Reactors? IE curseforge download, minecraft forums, and or Github links? And I hope things get better for @erogenousbeef

**Comment by Gunner76th** [comment]
**_Update**_
It looks as if Cyberdyne has dropped Big Reactors as well, as someone else beat them to a 1.10.2 release. 
https://github.com/CyberdyneCC/BigReactors/issues/1

The link for the now current mod author/dev for 1.10.2 is Zero no Ryouki and his forums thread on Big Reactors is located here: http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/mods-discussion/2693080-big-reactors-port-for-1-9-4-1-10-2-anyone

**Comment by XFactHD** [comment]
I am a member of the CybersyneCC, I know that we dropped the 1.10 support. But a direct port (which is what Zero no Ryouki did) can be done faster than a complete rewrite (which is what we wanted to do). This however doesn't mean, that we dropped 1.7.10 support.

**Comment by Gunner76th** [comment]
Just a point to note, my post stated nothing about your support for 1.7.10, only that you dropped support for 1.10.2. As most people are now starting to move over to the new 1.10.2 version now that forge is fully supported, and more mods have evolved to the new code, it only seemed appropriate to post those links for those looking for this new information. The only reason I found out about it is because of a thread on your github post. 

I would suggest updating the main page of your github to reflect this as Tomson24 is still declaring that CyberdyneCC is (and I quote) "Since minecraft 1.7.10 is already starting to die out we decided to concentrate on a version for 1.9.x." This is in direct contradition with the one and only issues post on the CyberdyneCC github for Big Reactors (which isnt a bug report but a features discussion) https://github.com/CyberdyneCC/BigReactors/issues/1 "Guys, it seems that some of you don't know it yet, but I quit the work on this mod as there is already another port in work which is much closer to releasing then mine was... "

Again, no where in my reply did I mention anything about 1.7.10, but bringing to light that a 1.10.2 port was out, and that it was not from CyberDyneCC but instead from a separate author.

**Comment by XFactHD** [comment]
Sorry, that was a bad assumption on my side. I am gonna look into this issue again and see what I can do, maybe I find the problem and we will be able to make a final version for 1.7.10.
